src/main/java/com/sprint/mission/discodeit/config/SecurityConfig.java

```diff
@@ -0,0 +1,291 @@
+import org.springframework.security.web.context.HttpSessionSecurityContextRepository;
+import org.springframework.security.web.csrf.CookieCsrfTokenRepository;
+import org.springframework.security.web.csrf.CsrfToken;
+import org.springframework.security.web.csrf.CsrfTokenRequestAttributeHandler;
+import org.springframework.security.web.csrf.CsrfTokenRequestHandler;
+import org.springframework.security.web.session.ConcurrentSessionFilter;
+import org.springframework.security.web.session.HttpSessionEventPublisher;
+import org.springframework.security.web.session.SessionInformationExpiredEvent;
+import org.springframework.security.web.session.SessionInformationExpiredStrategy;
+
+@Configuration
+@EnableWebSecurity
+@RequiredArgsConstructor
+@EnableMethodSecurity(prePostEnabled = true)
+@Slf4j
+public class SecurityConfig {
+
+    private final UserDetailsService userDetailsService;
+    private final UserMapper userMapper;
+    private final DataSource dataSource;
+    private final ObjectMapper objectMapper;
+
+    @Bean
+    public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
+        http
+                .csrf(csrf -> csrf
+                        .csrfTokenRepository(CookieCsrfTokenRepository.withHttpOnlyFalse())
+                        .csrfTokenRequestHandler(csrfTokenRequestHandler())
+                )
+                .securityContext(context -> context
+                        .securityContextRepository(new HttpSessionSecurityContextRepository())
+                )
+                .logout(logout -> logout
+                        .logoutUrl("/api/auth/logout")
+                        .addLogoutHandler(customLogoutHandler())
+                        .logoutSuccessHandler((request, response, authentication) -> {
+                            response.setStatus(HttpServletResponse.SC_OK);
+                            response.setContentType("application/json");
+                            response.getWriter().write("""
+                                        { "code": "LOGOUT_SUCCESS" }
+                                    """);
+                        })
+                )
+                .addFilterAt(jsonLoginFilter(authenticationManager()),
+                        UsernamePasswordAuthenticationFilter.class)
+                .addFilterBefore(concurrentSessionFilter(),
+                        UsernamePasswordAuthenticationFilter.class)
+                .authorizeHttpRequests(auth -> auth
+                        .requestMatchers(
+                                "/api/auth/login", "/api/users/**", "/api/auth/csrf-token",
+                                "/", "/index.html", "/swagger-ui/**", "/v3/api-docs/**",
+                                "/actuator/**", "/favicon.ico", "/assets/index-*.js",
+                                "/assets/index-*.css", "/static/**"
+                        ).permitAll()
+                        .anyRequest().authenticated()
```

Comment:
모든 HTTP 요청에 대해 인증 뿐만 아니라 인가처리도 해볼까요?  
```suggestion
                        .anyRequest().hasRole(Role.ROLE_USER.name())
```